File: i2f-jdk/i2f-code/src/main/java/i2f/code/CodeUtil.java
package i2f.code;


import java.security.SecureRandom;
import java.util.UUID;

/**
 * @author Ice2Faith
 * @date 2022/3/19 15:31
 * @desc
 */
public class CodeUtil {

    public static String makeUUID() {
        return UUID.randomUUID().toString().replace("-", "").toUpperCase();
    }

    public static volatile SecureRandom rand = new SecureRandom();

    public static String makeCheckCode(int len) {
        return makeCheckCode(len, false);
    }

    public static String makeCheckCode(int len, boolean onlyNumber) {
        String ret = "";
        int bounce = 10 + 26 + 26;
        if (onlyNumber) {
            bounce = 10;
        }
        for (int i = 0; i < len; i++) {
            int val = rand.nextInt(bounce);
            if (val < 10) {
                ret += (char) (val + '0');
            } else if (val < (10 + 26)) {
                ret += (char) (val - 10 + 'a');
            } else {
                ret += (char) (val - 10 - 26 + 'A');
            }
        }
        return ret;
    }
}

File: i2f-springboot/test-swl-starter/web/secure-web/src/secure/util/CodeUtil.js
import Random from './Random'

const CodeUtil = {
    makeCheckCode(len, onlyNumber = false) {
        let ret = ''
        let bounce = 10 + 26 + 26
        if (onlyNumber) {
            bounce = 10
        }
        for (let i = 0; i < len; i++) {
            const val = Random.nextLowerInt(bounce)
            if (val < 10) {
                ret += String.fromCharCode(val + '0'.charCodeAt())
            } else if (val < (10 + 26)) {
                ret += String.fromCharCode(val - 10 + 'a'.charCodeAt(0))
            } else {
                ret += String.fromCharCode(val - 10 - 26 + 'A'.charCodeAt(0))
            }
        }
        return ret
    }
}
export default CodeUtil
